app/server/lib/MinIOExternalStorage.ts

```diff
@@ -182,21 +189,38 @@ export class MinIOExternalStorage implements ExternalStorage {
   // Delete all versions of an object.
   public async _deleteAllVersions(key: string) {
     const vs = await this.versions(key, {includeDeleteMarkers: true});
-    await this._deleteBatch(key, vs.map(v => v.snapshotId));
+    await this._deleteVersions(key, vs.map(v => v.snapshotId));
   }
 
   // Delete a batch of versions for an object.
-  private async _deleteBatch(key: string, versions: Array<string | undefined>) {
+  private async _deleteVersions(key: string, versions: Array<string | undefined>) {
+    return this._deleteObjects(
+      versions.filter(v => v).map(versionId => ({
+        name: key,
+        versionId,
+      }))
+    );
+  }
+
+  // Delete an arbitrary number of objects, batched appropriately.
+  private async _deleteObjects(objects: { name: string, versionId?: string }[]): Promise<void> {
     // Max number of keys per request for AWS S3 is 1000, see:
     //   https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObjects.html
     // Stick to this maximum in case we are using this client to talk to AWS.
     const N = this._batchSize || 1000;
-    for (let i = 0; i < versions.length; i += N) {
-      const iVersions = versions.slice(i, i + N).filter(v => v) as string[];
-      if (iVersions.length === 0) { continue; }
-      await this._s3.removeObjects(this.bucket, iVersions.map(versionId => {
-        return { name: key, versionId };
-      }));
+    for (let i = 0; i < objects.length; i += N) {
+      const batch = objects.slice(i, i + N);
```

Comment:
what was the `filter(v => v)` in the old code?

Reply:
Not entirely sure - looks to be filtering out empty strings and undefined, but not sure which call site is passing those values down to here.

That check still exists anyway - all code which used to use `_deleteBatch`  uses `_deleteVersions` above, which does `filter(v => v)` before passing the values down to `_deleteObjects`.

`_deleteObjects` is only used by `removeAllWithPrefix` directly, which passes it the output from S3's `_listObjects`, so should all be valid?